### PR TITLE
Try fixing memory_profile_test on libuv iomgr

### PR DESCRIPTION
**Pull request by jtattermusch**
Tentative fix for one of the failures mentioned in https://github.com/grpc/grpc/issues/14926.

### CODE REVIEW

File: src/core/lib/iomgr/tcp_uv.cc [hunk 1]
```diff
@@ -356,7 +356,9 @@ static grpc_resolved_addresses* handle_addrinfo_result(
     i++;
     prev = resp;
     resp = resp->ai_next;
-    gpr_free(prev);
+    // addrinfo objects are allocated by libuv (e.g. in uv_getaddrinfo)
+    // and not by gpr_malloc
+    free(prev);
```
Comment (jtattermusch) [inline src/core/lib/iomgr/tcp_uv.cc]: Please doublecheck, I'm not familiar with the libuv code.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
This seems to indeed fix the `bins/dbg/memory_profile_test` (tested locally).

File: src/core/lib/iomgr/tcp_uv.cc (hunk 1 above)
Comment (kpayson64) [reply src/core/lib/iomgr/tcp_uv.cc]: If this is uv allocated memory, we might want to go all the way and just skip the free here and at the end of the function
call ```uv_freeaddrinfo(result)```

**Comment by jtattermusch** [comment]
Update to use  `uv_freeaddrinfo(result)` instead. PTAL.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by murgatroid99** [comment]
It looks like the original implementation freed that memory using `uv_freeaddrinfo`, so I think that confirms that this change is correct. Also, this should fix a bug I'm seeing, so I'm just going to merge now.

**Pull request merged**